resources/sts/4.14/hypershift/sts_extended_hcp_support_permission_policy.json

```diff
@@ -2,58 +2,133 @@
+
+            ],
+            "Resource": "arn:aws:ec2:*:*:snapshot/*"
+        },
+        {
+            "Sid": "SendSerialConsoleSSHPublicKey",
+            "Effect": "Allow",
+            "Action": [
+                "ec2-instance-connect:SendSerialConsoleSSHPublicKey",
+                "ec2:GetSerialConsoleAccessStatus"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "DescribeNetworkInsightsPath",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeNetworkInsightsPaths",
+                "ec2:StartNetworkInsightsAnalysis"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "NetworkInsightsPath",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DeleteNetworkInsightsPath"
+            ],
+            "Resource": "arn:aws:ec2:*:*:network-insights-path/*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
                 "ec2:CreateNetworkInsightsPath",
-                "ec2:CreateSnapshot",
-                "ec2:CreateSnapshots",
-                "ec2:DeleteNetworkInsightsAnalysis",
-                "ec2:DeleteNetworkInsightsPath",
-                "ec2:DescribeAccountAttributes",
-                "ec2:DescribeAddresses",
-                "ec2:DescribeAddressesAttribute",
-                "ec2:DescribeAggregateIdFormat",
-                "ec2:DescribeByoipCidrs",
-                "ec2:DescribeCapacityReservations",
-                "ec2:DescribeCarrierGateways",
-                "ec2:DescribeClassicLinkInstances",
+                "ec2:DescribeNetworkInsightsAnalyses"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "NetworkInsightsAnalysis",
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DeleteNetworkInsightsAnalysis"
+            ],
+            "Resource": "arn:aws:ec2:*:*:network-insights-analysis/*"
+        },
+        {
+            "Sid": "ClientVpn",
+            "Effect": "Allow",
+            "Action": [
                 "ec2:DescribeClientVpnAuthorizationRules",
                 "ec2:DescribeClientVpnConnections",
-                "ec2:DescribeClientVpnEndpoints",
                 "ec2:DescribeClientVpnRoutes",
-                "ec2:DescribeClientVpnTargetNetworks",
-                "ec2:DescribeCoipPools",
-                "ec2:DescribeCustomerGateways",
-                "ec2:DescribeEgressOnlyInternetGateways",
-                "ec2:DescribeIdentityIdFormat",
-                "ec2:DescribeIdFormat",
-                "ec2:DescribeImageAttribute",
-                "ec2:DescribeImages",
+                "ec2:DescribeClientVpnTargetNetworks"
+            ],
+            "Resource": "arn:aws:ec2:*:*:client-vpn-endpoint/*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeClientVpnEndpoints"
+            ],
+            "Resource": "*"
```

Comment:
Can this be restricted at all?

Reply:
Can't because i have tried running with restriction 	"Resource": "arn:aws:ec2:*:*:client-vpn-endpoint/*" and it failed. 
`
aws ec2 describe-client-vpn-endpoints --dry-run
An error occurred (UnauthorizedOperation) when calling the DescribeClientVpnEndpoints operation: You are not authorized to perform this operation. User: arn:aws:sts::301721915996:assumed-role/sdq-ci-tgsde-HCP-ROSA-Support-Role/5a9d529885b8d55fb4b1902fd87f9c2f is not authorized to perform: ec2:DescribeClientVpnEndpoints because no identity-based policy allows the ec2:DescribeClientVpnEndpoints action
—-> change resource to /*
`